game/ffmpeg.cc

```diff
@@ -277,6 +300,46 @@ FFmpeg::FFmpeg(fs::path const& _filename, int mediaType) : m_filename(_filename)
 	m_codecContext = std::move(pCodecCtx);
 }
 
+
+/**
+  * \brief    Fetch the Replay Gain "loudness" factor from the stream
+  *
+  * \details  Replay Gain is a decibels "loudness" factor that can be used to normalise
+  *           the human-perceived loudness of a given audio sample.  It is useful for
+  *           normalising the volume of recordings mastered at different volumes.
+  *           Libav reads this information in, storing it in the side channel blocks.
+  *           This function requests this value from the side channel.
+  *
+  *           Ref: https://en.wikipedia.org/wiki/ReplayGain
+  *                https://www.smithsonianmag.com/smart-news/music-does-get-louder-every-year-835681/
+  *
+  * \note     The libav function av_stream_get_side_data() has been deprecated
+  *           It will need to be updated along with all the others
+  *
+  * \returns  True if a Replay Gain value was found in the stream
+  */
+bool FFmpeg::readReplayGain(const AVStream *stream)
+{
+    bool rg_read = false;
+    // assert(stream)
+    m_replaygain = 0.0;
+    if (stream != nullptr) {
+        size_t replay_gain_size = 0;
+        const AVReplayGain *replay_gain = (AVReplayGain *)av_stream_get_side_data(stream, AV_PKT_DATA_REPLAYGAIN, &replay_gain_size);
+        if (replay_gain_size > 0 && replay_gain != nullptr) {
+            m_replaygain = static_cast<double>(replay_gain->track_gain);
+            m_replaygain /= 100000.0;   // convert from microbels to decibels
```

Comment:
If you store micro decibel, name the variable `m_replaygainInMicroDecibel`.

Reply:
But I think it should be stored in decibel. Where you need micro decibel convert it.